packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -460,6 +460,11 @@ export class EndpointAccess {
    * @param cidr CIDR blocks.
    */
   public onlyFrom(...cidr: string[]) {
+    if (!this._config.privateAccess) {
+      // when private access is disabled, we can't restric public
+      // access since it will render the kubectl provider unusable.
+      throw new Error('Cannot restric public access to endpoint when private access is disabled. Use PUBLIC_AND_PRIVATE.onlyFrom() instead.');
```

Comment:
The terminology "restrict" is a bit vague. I would explicitly say something like "cannot specify cidr blocks for public access..."